### PR TITLE
fix: app should only have one instance

### PR DESCRIPTION
**Pull request by pixelass**

### CODE REVIEW

File: src/electron/background.ts [hunk 1]
```diff
@@ -56,21 +57,36 @@ import "@/ipc/vector-store";
...
-
-// Listen for the 'window-all-closed' event on the Electron app object.
-// This event is emitted when all windows of the application have been closed.
-// In response, quit the application to free up resources, adhering to typical desktop application
-// behavior.
-app.on("window-all-closed", () => {
+
+if (gotTheLock || isTest) {
+	// Initialize the application by calling the main function.
+	// Upon completion, log to the console indicating the application has started.
```
Comment (TimPietrusky) [inline src/electron/background.ts]:
```suggestion
```

Comment (TimPietrusky) [reply src/electron/background.ts]: Comment seems useless, as its clear that something is logged to the console.

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
I just have a super minor change, everything else looks perfect! Thank you.

File: src/electron/background.ts (hunk 1 above)
Comment (pixelass) [reply src/electron/background.ts]: this was not a change of the PR though. 

But I think the flag here is not actually needed anymore as it was just aiming to fix the broken tests. I later identified the actual root cause.

I will remove the flag condition and remove the comment along the way.

Comment (pixelass) [reply src/electron/background.ts]: seems like the flag was actually needed afterall. Comments were removed

**Review by TimPietrusky — APPROVED** [review]

**Pull request merged**